plugins/onepad/Linux/dialog.cpp

```diff
@@ -391,11 +391,14 @@ void on_view_key_clicked(GtkToggleButton *togglebutton, gpointer user_data)
 void on_toggle_option(GtkToggleButton *togglebutton, gpointer user_data)
 {
 	dialog_checkbox *checkbox = (dialog_checkbox*)user_data;
+	u32 joyid = conf->get_joyid(current_pad);
 
 	if (gtk_toggle_button_get_active(togglebutton))
 		conf->options |= checkbox->mask;
 	else
 		conf->options &= ~checkbox->mask;
+
+	s_vjoysticks[joyid]->Init(joyid);
```

Comment:
Add me a comment here to explain that it is done to reload the joystick to take into account new option such as DS3 hack.